extension/src/parsing/dclParser.ts

```diff
@@ -0,0 +1,271 @@
+		this.saveTempData();
+		this.saveCurrentAtom(this.data.charAt(this.dataIndex));
+		return this;
+	}
+
+	createNewTileScopeFromFragments(): void {
+		const tile = new DclTile(this.linefeed, this.fragments);
+		this.activeContainer.atoms.push(tile);
+		this.containers.push(tile);
+		this.fragments.length = 0;
+		this.containerIndex++;
+	}
+
+	closeActiveTileScope(): void {
+		this.fragments.forEach(x => {			
+			this.activeContainer.atoms.push(x);
+		});
+		this.fragments.length = 0;		
+		this.containerIndex--;
+		this.containers.pop();
+	}
+
+	storeChar(charCode: number) : DclDocContext {
+		if (!this.temp.hasValues()) {
+			this.startGroupPointer();
+		}
+		this.temp.appendCode(charCode);
+		return this;
+	}
+
+
+	saveFragmentsAsAttribute() : DclDocContext {
+		// Forces comments into the parent Tile Container and creates Attribute from remaining elements. This only 
+		// needs to inspect the head for comments because encountering a comment causes an attribute accumulation event.
+		let skip = 0;
+		this.fragments.forEach(atom => {
+			if (atom.isComment) {
+				this.activeContainer.atoms.push(atom);
+				skip++;
+			}
+		});
+		if (this.fragments.length - skip > 0) {
+			this.activeContainer.atoms.push(new DclAttribute(this.fragments.slice(skip)));
+		}
+		this.fragments.length = 0;
+		return this;
+	}
+
+	processCommentAndSaveAsFragment(isBlock: boolean): void {
+		while (this.dataIndex < this.data.length) {
+			const curr = this.data.charCodeAt(this.dataIndex);
+			const next = this.data.charCodeAt(this.dataIndex + 1); 
+			this.storeChar(curr)
+				.incrementAtLineBreak(curr) // does nothing when (curr !== lf)
+			    .moveNext();
+			if (isBlock && curr === asterisk && next === forwardSlash) {
+				this.storeChar(next)
+					.moveNext();
+				break;
+			} else if (!isBlock && (next === cr || next === lf)) {
+				break;
+			}
+		}	
+		this.saveTempData();
+	}
+
+	processStringAndSaveAsFragment(): void {
+		this.storeChar(dblQuote)
+		    .moveNext();
+		let isEscaping = false;
+		
+		while (this.dataIndex < this.data.length) {
+			const curr = this.data.charCodeAt(this.dataIndex);
+			this.storeChar(curr)
+			    .moveNext();
+			if (curr === backSlash) {
+				isEscaping = !isEscaping;
+			} else if (curr === dblQuote && !isEscaping) {
+				break;
+			} else if (isEscaping) {
+				isEscaping = false;
+			}
+		}	
+		this.saveTempData();
+	}
+
+}
+
+
+
+
+function _getDclDocumentContainer(ctx: DclDocContext) {	
+	while (ctx.dataIndex < ctx.data.length) {
+		const curr = ctx.data.charCodeAt(ctx.dataIndex);
+		const next = ctx.data.charCodeAt(ctx.dataIndex + 1); 
+
+		if (curr === openBracket) {	// tile/attribute breakpoint
+			ctx.saveTempDataAndCurrentCharToFragments()
+			   .moveNext()
+			   .createNewTileScopeFromFragments();
+		} else if (curr === closeBracket) { // tile/attribute breakpoint
+			ctx.saveTempData()
+			   .saveFragmentsAsAttribute()
+			   .saveTempDataAndCurrentCharToFragments()
```

Comment:
This looks sketchy, but is doing what is intended. If you want, I can go create a `saveCurrentCharToFragments` function so this doesn't look redundant.